hack/build-image.sh

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env bash
+
+set -eu
+
+# Warn when unprivileged
+if [ `id --user` -ne 0 ]; then
+	echo "---"
+	echo "Note: Building unprivileged may fail due to permissions"
+	echo "---"
+fi
+
+# Record all images files
+ALL_IMAGES=""
+for x in `ls Dockerfile.*`; do
+	ALL_IMAGES="$ALL_IMAGES$x "
+done
+
+# To build will hold a list of all image files to build
+TOBUILD=""
+
+# WHAT should be defined. If not, give a list and exit
+if [ -z ${WHAT+a} ]; then
+	echo "WHAT must be set to one of the following:"
+	echo "- all"
+	for x in $ALL_IMAGES ; do
+		echo "-" `echo $x | sed -e "s|Dockerfile.||"`
+	done
+
+	exit 1
+fi
+
+
+# If all is the WHAT target then set TOBUILD to all the images found
+if [ ${WHAT} == "all" ]; then
+	TOBUILD=$ALL_IMAGES
+else
+	# Otherwise WHAT should be valid at this point
+	TOBUILD=Dockerfile.${WHAT}
+fi
+
+# Check that the target is valid
+if [[ $ALL_IMAGES != *${TOBUILD}* && ${WHAT} != "all" ]]; then
+	echo "${WHAT} is not a valid build target"
+	exit 1
+fi
+
+if [ -z ${VERSION+a} ]; then
+        echo "Using version from git..."
+        VERSION=$(git describe --abbrev=8 --dirty --always)
+fi
+
+# Build all images requested
+for IMAGE_TO_BUILD in $TOBUILD; do
+	NAME=`echo $IMAGE_TO_BUILD | sed -e "s|Dockerfile.||"`
+	CMD="podman build -t $NAME:$VERSION -f $IMAGE_TO_BUILD"
```

Comment:
add info in README and mention the script requires `podman`

Reply:
Washing through `$CMD` will break any quoting to protect from an invalid `${WHAT}`.  If we want to echo the commands being executed, we can, since we're already using Bash, use:

```sh
set -x
for IMAGE_TO_BUILD in $TOBUILD; do
  # ...test for file existence, see my earlier comment...
  podman build -t "${NAME}:${VERSION}" -f "${IMAGE_TO_BUILD}"
done
```